tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.test;
+
+import org.ballerinalang.model.symbols.SymbolKind;
+import org.ballerinalang.model.symbols.SymbolOrigin;
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.semantics.analyzer.Types;
+import org.wso2.ballerinalang.compiler.semantics.model.Scope;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BTypeDefinitionSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BType;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static io.ballerina.test.utils.Constants.COMMENT;
+import static io.ballerina.test.utils.Constants.DISABLED_FILE;
+import static io.ballerina.test.utils.Constants.FAILING_FILE;
+import static io.ballerina.test.utils.Constants.SUBTYPE_SYMBOL;
+
+/**
+ * Test semantic type relations.
+ * The notation T1<:T2 means T1 is a subtype of T2.
+ *
+ * @since 2.0.0
+ */
+public class SemTypeTest {
+
+    @DataProvider(name = "filePathProvider")
+    public Object[] filePathProvider() {
+        File dir = resolvePath("test-src").toFile();
+        List<String> files = new ArrayList<>();
+        for (File file : Objects.requireNonNull(dir.listFiles())) {
+            String fileName = file.getName();
+            if (fileName.contains(FAILING_FILE) || fileName.contains(DISABLED_FILE)) {
+                continue;
+            }
+            files.add(file.getAbsolutePath());
+        }
+        return files.toArray();
+    }
+
+    @Test(dataProvider = "filePathProvider")
+    public void testSubTypeRelationship(String filePath) throws IOException {
+        List<String> actualRelations = actualSubTypeRelations(filePath);
```

Comment:
I have a feeling that `actualRelations` and `expectedRelations` should be `Set<String>`.

Reply:
@rdhananjaya, is there any specific reason to do this?

Reply:
We only need set difference between expected and calculated relations.

Reply:
Changed